api/test/unit/services/application.service.spec.ts

```diff
@@ -812,1750 +812,2012 @@ describe('Testing application service', () => {
-        },
-        preferredUnitTypes: {
-          connect: [
+          preferredUnitTypes: {
+            connect: [
+              {
+                id: expect.anything(),
+              },
+            ],
+          },
+          householdMember: {
+            create: [
+              {
+                orderId: 0,
+                firstName: 'example first name',
+                middleName: 'example middle name',
+                lastName: 'example last name',
+                birthMonth: 12,
+                birthDay: 17,
+                birthYear: 1993,
+                sameAddress: YesNoEnum.yes,
+                relationship: HouseholdMemberRelationship.other,
+                workInRegion: YesNoEnum.yes,
+                householdMemberAddress: {
+                  create: {
+                    ...exampleAddress,
+                  },
+                },
+                householdMemberWorkAddress: {
+                  create: {
+                    ...exampleAddress,
+                  },
+                },
+              },
+            ],
+          },
+          programs: [
+            {
+              key: 'example key',
+              claimed: true,
+              options: [
+                {
+                  key: 'example key',
+                  checked: true,
+                  extraData: [
+                    {
+                      type: InputType.boolean,
+                      key: 'example key',
+                      value: true,
+                    },
+                  ],
+                },
+              ],
+            },
+          ],
+          preferences: [
             {
-              id: expect.anything(),
+              key: 'example key',
+              claimed: true,
+              options: [
+                {
+                  key: 'example key',
+                  checked: true,
+                  extraData: [
+                    {
+                      type: InputType.boolean,
+                      key: 'example key',
+                      value: true,
+                    },
+                  ],
+                },
+              ],
             },
           ],
+          userAccounts: {
+            connect: {
+              id: 'requestingUser id',
+            },
+          },
+        },
+      });
+
+      expect(canOrThrowMock).toHaveBeenCalledWith(
+        {
+          id: 'requestingUser id',
+          userRoles: { isAdmin: true },
+        } as unknown as User,
+        'application',
+        permissionActions.create,
+        {
+          listingId: dto.listings.id,
+          jurisdictionId: expect.anything(),
         },
-        householdMember: {
-          create: [
-            {
-              orderId: 0,
-              firstName: 'example first name',
-              middleName: 'example middle name',
-              lastName: 'example last name',
+      );
+    });
+
+    it('should create an application from partner site when listing is closed', async () => {
```

Comment:
This is the new test added. The rest of this file is just to better organize the file by endpoint